src/api_tracker.py

```diff
@@ -0,0 +1,68 @@
+import os
+import requests
+from ckan.common import config
+import ckan.plugins.toolkit as tk
+import hashlib
+import logging
+import re
+
+
+log = logging.getLogger(__name__)
+
+def ga_api_tracker(resource_id):
+    try:
+        tid = config.get('googleanalytics.id', None)
+        endpoint = config.get('googleanalytics.endpoint', None)
+        client_id = hashlib.md5(tk.c.user).hexdigest()
+        payload = {
+            'v': 1, # Version.
+            'tid': tid, # Tracking ID
+            'cid': client_id, # Anonymous Client ID
+            't': 'event', # Hit type (event, pageview, etc)
+            'ec': 'Resource', # Event Category
+            'ea': 'API', # Event Action
+            'el': resource_id # Event label
+        }
+        log.warning('sending payload to GA: {}'.format(payload))
+        r = requests.post(endpoint, data=payload, headers={'User-Agent': 'User Agent 1.0'})
+        if r.ok:
+            log.info('ga for search api sent')
+        else:
+            log.info('ga for search api failed')
+    except Exception as e:
+        log.error(str(e))
+
+def ga_search_sql_api_tracker(sql):
+    try:
+        # try to extract table name from sql statement
+        table = tables_in_query(sql)
+        log.warning('table name: {}'.format(table))
+        if (table):
+            ga_api_tracker(table)
+    except Exception as e:
+        log.error(str(e))
+
+
+def tables_in_query(sql_str):
```

Comment:
i thought this method exists in core ckan in datastore code ...

Reply:
do you mean [resource_id_from_alias](https://github.com/ckan/ckan/blob/1101c3b72396adf9d4d75dafc968e81b031cdbc8/ckanext/datastore/backend/postgres.py#L1939)?

Reply:
No, i mean https://github.com/ckan/ckan/blob/1101c3b72396adf9d4d75dafc968e81b031cdbc8/ckanext/datastore/helpers.py#L73

Reply:
this method working against postgresql, that is not our case here

Reply:
ok got you.